bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -1628,12 +1596,28 @@ void ReplicaImp::onInternalMsg(InternalMessage &&msg) {
         ccss->seqNumber, ccss->view, ccss->combinedSig.data(), ccss->combinedSig.size(), ccss->span_context_);
   }
   if (auto *ccsf = std::get_if<CombinedCommitSigFailedInternalMsg>(&msg)) {
-    return onCommitCombinedSigFailed(ccsf->seqNumber, ccsf->view, ccsf->replicasWithBadSigs);
+    return onCommitCombinedSigFailed(ccsf->seqNumber, ccsf->view, CommitPath::SLOW, ccsf->replicasWithBadSigs);
   }
   if (auto *vccs = std::get_if<VerifyCombinedCommitSigResultInternalMsg>(&msg)) {
     return onCommitVerifyCombinedSigResult(vccs->seqNumber, vccs->view, vccs->isValid);
   }
 
+  // Handle Fast Path Commit related internal messages
+  if (auto *ccss = std::get_if<FastPathCombinedCommitSigSucceededInternalMsg>(&msg)) {
+    return onFastPathCommitCombinedSigSucceeded(ccss->seqNumber,
+                                                ccss->view,
```

Comment:
Should we continue calling this function if currentView() != ccss->view?

Reply:
This check is done inside onFastPathCommitCombinedSigSucceeded similar to onCommitCombinedSigSucceeded. Should we handle it differently?